src/titiler/xarray/titiler/xarray/io.py

```diff
@@ -138,27 +138,37 @@ def _arrange_dims(da: xarray.DataArray) -> xarray.DataArray:
 def get_variable(
     ds: xarray.Dataset,
     variable: str,
-    datetime: Optional[str] = None,
-    drop_dim: Optional[List[str]] = None,
+    sel: Optional[List[str]] = None,
+    method: Optional[Literal["nearest", "pad", "ffill", "backfill", "bfill"]] = None,
 ) -> xarray.DataArray:
     """Get Xarray variable as DataArray.
 
     Args:
         ds (xarray.Dataset): Xarray Dataset.
         variable (str): Variable to extract from the Dataset.
-        datetime (str, optional): datetime to select from the DataArray.
-        drop_dim (list of str, optional): DataArray dimension to drop in form of `[{dimension}={value}],`.
+        sel (list of str, optional): List of Xarray Indexes.
+        method (str): Xarray indexing method.
 
     Returns:
         xarray.DataArray: 2D or 3D DataArray.
 
     """
     da = ds[variable]
 
-    if drop_dim:
-        for dim in drop_dim:
-            dim_to_drop, dim_val = dim.split("=")
-            da = da.sel({dim_to_drop: dim_val}).drop_vars(dim_to_drop)
+    if sel:
+        sel_idx = []
+        for s in sel:
+            val: Union[str, slice]
+            dim, val = s.split("=")
+            dt = da[dim].dtype
+            if "," in val:
+                val = slice(*[numpy.array(v).astype(dt) for v in val.split(",")])
+            else:
+                val = numpy.array(val).astype(dt)
```

Comment:
we need to `cast` the value to the array datatype (e.g `z="0"` -> `z=0`)

Reply:
Do we? in the `slice` case, yes. But otherwise, I think Xarray / Pandas does this casting for you, no?

Reply:
Does this work well for datetime dtype?

For datetime dimension dtype, you could also not cast the indices (do the comma-split to slices, though) and leave the casting to Pandas, preserving the nice magic is has: https://github.com/developmentseed/titiler/issues/1112#issuecomment-2761260478

Reply:
🤦 we need to do like xpublish https://github.com/xpublish-community/xpublish-wms/blob/15aaa48ead7468f80d4794456b674a28bbc7dbe0/xpublish_wms/wms/get_map.py#L251-L269